src/v/utils/token_bucket.h

```diff
@@ -62,7 +73,21 @@ class token_bucket {
             _refresh_timer.arm(refresh_interval - elapsed);
         }
 
-        return _sem.wait(as, size);
+        auto deficiency = size - _sem.available_units();
```

Comment:
did you mean `available_units` here vs one of the other semaphore interfaces for querying units? i ask because this one can become negative depending on how the semaphore is used.

Reply:
in this context it should go negative but still, I think that the deficiency value could be incorrect in case of concurrency, I'll update this